draft-pauly-add-deer.md

```diff
@@ -170,44 +170,49 @@ Encrypted Resolver name.
 
 ## Authenticated Discovery {#authenticated}
 
+If the Unencrypted Resolver's IP address is in a public range, it is considered
+a global reference identity, and is subject to cryptographic authentication.
+
 In order to be considered an authenticated Equivalent Encrypted Resolver, the
-TLS certificate presented by the Encrypted Resolver MUST contain both the domain
-name (from the SVCB answer) and the IP address of its equivalent Unencrypted
+TLS certificate presented by the Encrypted Resolver MUST contain the IP address
+of its equivalent Unencrypted
 Resolver within the SubjectAlternativeName certificate field. The client MUST
-check the SubjectAlternativeName field for both the Unencrypted Resolver's IP
-address and the advertised name of the Equivalent Encrypted Resolver. If the
+check the SubjectAlternativeName field for the Unencrypted Resolver's IP
+address. If the
 certificate can be validated, the client SHOULD use the discovered Equivalent
 Encrypted Resolver for any cases in which it would have otherwise used the
-Unencrypted Resolver. If the Equivalent Encrypted Resolver has a different IP
-address than the Unencrypted Resolver and the TLS certificate does not cover the
-Unencrypted Resolver address, the client MUST NOT use the discovered Encrypted
+Unencrypted Resolver.  Otherwise, the client MUST NOT use the discovered Encrypted
 Resolver. Additionally, the client SHOULD suppress any further queries for
 Equivalent Encrypted Resolvers using this Unencrypted Resolver for the length of
 time indicated by the SVCB record's Time to Live (TTL).
 
-If the Equivalent Encrypted Resolver and the Unencrypted Resolver share an IP
-address, clients MAY choose to opportunistically use the Encrypted Resolver even
-without this certificate check ({{opportunistic}}).
-
 ## Opportunistic Discovery {#opportunistic}
 
-There are situations where authenticated discovery of encrypted DNS
-configuration over unencrypted DNS is not possible. This includes Unencrypted
-Resolvers on non-public IP addresses whose identity cannot be confirmed using
-TLS certificates.
+If the Unencrypted Resolver's IP address is in a private range, it is considered
+a local reference identity that cannot be confirmed using TLS certificates.
+Instead, the ability to respond to queries confirms control of the address.
+The client MUST repeat the discovery process at least every 5 minutes (e.g. by
```

Comment:
Would this  be a requirement only when the IP address is different from the original unencrypted resolver address? If the private address is able to host the DoH server directly, I wonder if this freshness matters as much.

Reply:
I think it would be OK to relax this when the addresses match, but I lean toward avoiding the special-case.

Theoretically, the question is: does the presence of a working Encrypted Resolver on the same IP mean that it is the intended EER?  This seems like a reasonable assumption, but it could be false if the Unencrypted Resolver IP hosts multiple DoH paths with different behaviors, or if it offers DoT on port 853 but would prefer to direct users to an EER somewhere else for load balancing, etc.  In these cases, an attacker could inject a DEER response that contradicts the Unencrypted Resolver's intent, while keeping the same IP.

Keeping the time limit across the board seems easier to implement, and also provides some defense against these (admittedly marginal) attacks.

Reply:
I agree that less complex specs are better in general. I also agree that for home users on personal Wi-Fi with <100 devices this makes total sense.

However, I think the situation will be different for enterprises with tens of thousands of devices. Large corporate networks may choose to deploy encrypted DNS on the same IP address as they already use for unencrypted DNS. For many devices, they may only ever use DEER in this context, and having the network flooded with these polls would be painful. They would know if they offer different encrypted services on the same address (unlikely) and may want to accept those minor risks they are capable of managing in a managed environment.

To that end, I would like to see the polling for same-local-address made optional, even if we clearly highlight the risks and recommend checking anyway to enable standard behavior for clients who know it works better for their situation.

Reply:
OK, I've added an "Optimizations" section that mentions this optimization, and a few others as well.  I've also adjusted the security considerations.

Reply:
LGTM. The guidance regarding polling is a nice touch.